spec/game_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "tictactoe/board"
+require "tictactoe/board_options"
+require "tictactoe/game_type_options"
+require "web_player_factory"
+require "game_controller"
+
+RSpec.describe GameController do
+  let(:hvh_option) { 1 }
+  let(:hvh) { TicTacToe::GameTypeOptions::ID_TO_GAME_TYPE[hvh_option] }
+  let(:dimension) { TicTacToe::BoardOptions::DIMENSIONS["THREE_BY_THREE"] }
+  let(:display) { WebDisplay.new }
+  let(:players) { Hash[TicTacToe::Mark::X, WebHumanPlayer.new(TicTacToe::Mark::X, display), TicTacToe::Mark::O, WebHumanPlayer.new(TicTacToe::Mark::O, display)] }
+  let(:player_factory_spy) { instance_spy(WebPlayerFactory) }
+
+  it "creates a new hvh, 3x3 game" do
+    expect(player_factory_spy).to receive(:get_players_for_game_type).with(hvh_option).and_return(players)
+    game_controller = GameController.new(player_factory_spy)
+    game_controller.create_game(dimension, hvh_option, display)
+    expect(game_controller.game_type).to be(hvh_option)
+    expect(game_controller.board.board_size).to be(dimension*dimension)
+  end
+
+  it "plays a move" do
+    expect(player_factory_spy).to receive(:get_players_for_game_type).with(hvh_option).and_return(players)
```

Comment:
same assertion as previous test which is duplication in the tests. you could use allow to stub it instead of a self-verifying mock.